driver/src/main/com/mongodb/ChangeStreamIterableImpl.java

```diff
@@ -96,45 +98,49 @@
     }
 
     @Override
-    public MongoCursor<TResult> iterator() {
-        return execute().iterator();
+    public <TDocument> MongoIterable<TDocument> rawResult(final Class<TDocument> clazz) {
+        return execute(codecRegistry.get(clazz));
```

Comment:
It's a bit unexpected that this method's implementation (and map's too, now that I look at it, actually starts execution.  That's different behavior that FindIterableImpl.map.  

Can we delay execution?  For map at least, it's easy:

        return new MappingIterable<ChangeStreamOutput<TResult>, U>(this, mapper);
    
For rawResult, it may require making the type of the codec property fully generic.